src/plugins/intel_gpu/src/plugin/transformations_pipeline.cpp

```diff
@@ -221,27 +176,77 @@ void TransformationsPipeline::apply(std::shared_ptr<ov::Model> func) {
         for (auto& et : fp_element_types) {
             if (!fp_precision_supported(et)) {
                 auto et_pair = std::make_pair(et, fallback_precision);
-                bool has_valid_conversion = std::find_if(convert_precision_list.begin(), convert_precision_list.end(),
-                    [&](std::pair<ov::element::Type, ov::element::Type> v) -> bool {
-                        return v.first == et_pair.first && fp_precision_supported(v.second);
-                }) != convert_precision_list.end();
+                bool has_valid_conversion = std::find_if(fp_convert_precision_list.begin(), fp_convert_precision_list.end(),
+                                                         [&](std::pair<ov::element::Type, ov::element::Type> v) -> bool {
+                                                             return v.first == et_pair.first && fp_precision_supported(v.second);
+                                                         }) != fp_convert_precision_list.end();
 
                 if (!has_valid_conversion) {
-                    convert_precision_list.push_back(et_pair);
+                    fp_convert_precision_list.push_back(et_pair);
                 }
             }
         }
 
+        type_to_fuse_map empty_fuse_map = {};
         manager.register_pass<ngraph::pass::Validate>();
-        manager.register_pass<ngraph::pass::ConvertPrecision>(convert_precision_list);
+        //  call ConvertPrecision with keep_precision_sensitive_in_fp32 = true
+        manager.register_pass<ngraph::pass::ConvertPrecision>(fp_convert_precision_list, empty_fuse_map, true);
+
+        precisions_array int_and_f64_convert_precision_list {
+                {ngraph::element::f64, ngraph::element::f32},
+                {ngraph::element::i64, ngraph::element::i32},
+                {ngraph::element::u64, ngraph::element::i32},
+                {ngraph::element::u16, ngraph::element::i32},
+                {ngraph::element::u32, ngraph::element::i32},
+                {ngraph::element::boolean, ngraph::element::u8},
+                {ngraph::element::i4, ngraph::element::i8},
+                {ngraph::element::u4, ngraph::element::u8},
+        };
+
+        manager.register_pass<ngraph::pass::Validate>();
+        manager.register_pass<ngraph::pass::ConvertPrecision>(int_and_f64_convert_precision_list);
```

Comment:
I think we wanted to keep these converts after common optimizations (as it is now) to minimize risk of breaking something else.

Reply:
Yes, you are right. At the moment when I was preparing PR I moved marking inside `ConvertPrecision` and float conversion from `CommonOptimizations` into `transformations_pipeline` it seemed not a big deal to move integer conversion as well. But you are right, let's be consistent with what we discussed. I'll move integer and int64 conversion back after `CommonOptimizations`

Reply:
Returned back integer conversion where it was at current master: to the end of CommonOptimizations. Please take a look